Slim/Formats/Playlists/Base.pm

```diff
@@ -39,6 +40,13 @@ sub _updateMetaData {
 		}
 	}
 	
+	# Treat radio stream as playlist, because that's what they are.
+	# If we don't do this, we'll end up attempting to play the served m3u.
+
+	if ($entry =~ m{^http://opml\.(?:radiotime|tunein)\.com}) {
```

Comment:
> That would only help with stations played from the Radio menu, but no other stations.

That's correct, but I'm not sure what other information one could use to restore the 'playlist' flag. Any other ideas?

> Is this change even required?

IIRC (it's been a few years since I initially wrote this) the issue is that when restoring the playlist after power-off, the radio station M3U was restored as normal track, not as playlist, thus the M3U was attempted to be played instead of the MP3 stream it points to. So yes, I'm pretty sure it's required. I can try to gather the sequence of events without this piece again, though.

> And then the regex should at least match https, too ;-)

Good point, will update accordingly :-)

Reply:
https is now covered in the regex. Please let me know if you got any better idea for the playlist flag thing.